File: ios/Packages/Primitives/Sources/PaymentData.swift
// Copyright (c). Gem Wallet. All rights reserved.

import Foundation

public struct PaymentData: Hashable, Sendable {
    public let quote: PaymentQuote
    public let merchant: PaymentMerchant

    public init(quote: PaymentQuote, merchant: PaymentMerchant) {
        self.quote = quote
        self.merchant = merchant
    }
}

File: android/gemcore/src/main/kotlin/com/gemwallet/android/model/PaymentData.kt
package com.gemwallet.android.model

import com.wallet.core.primitives.PaymentMerchant
import com.wallet.core.primitives.PaymentQuote
import kotlinx.serialization.Serializable

@Serializable
data class PaymentData(
    val quote: PaymentQuote,
    val merchant: PaymentMerchant,
)
